server/routes/profile/middleware/retrieveConditions.ts

```diff
@@ -1,18 +1,36 @@
 import { NextFunction, Request, RequestHandler, Response } from 'express'
-import { ConditionService } from '../../../services'
+import { ConditionService, PrisonService } from '../../../services'
 import { Result } from '../../../utils/result/result'
+import mapPrisonNamesToReferencedAndAuditable from '../../../data/mappers/mapPrisonNamesToReferencedAndAuditable'
 
 /**
  *  Function that returns a middleware function to retrieve the prisoner's Conditions
  */
-const retrieveConditions = (conditionService: ConditionService): RequestHandler => {
+const retrieveConditions = (conditionService: ConditionService, prisonService: PrisonService): RequestHandler => {
   return async (req: Request, res: Response, next: NextFunction) => {
     const { prisonNumber } = req.params
     const { username } = req.user
 
     // Lookup the prisoner's Conditions and store in res.locals
     const { apiErrorCallback } = res.locals
-    res.locals.conditions = await Result.wrap(conditionService.getConditions(username, prisonNumber), apiErrorCallback)
+
+    const retrieveConditionsPromise = new Promise((resolve, reject) => {
```

Comment:
Horrible promise callback hell ....
The problem is that we have to return a promise in the `Result.wrap` construct (that's how that works)
Previously that was fine, in that we simply returned the `conditionService.getConditions` promise

But now we need to also call the prisons lookup (which is a promise), then wait for both to be resolved, then do the mapping; and then return the result as a resolved promise!

Yar! There be dragons!